zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -492,6 +663,18 @@ object ZQuery {
   def fromEffect[R, E, A](effect: ZIO[R, E, A]): ZQuery[R, E, A] =
     ZQuery(effect.foldCause(Result.fail, Result.done).provideSome(_._1))
 
+  /**
+   * Constructs a query from an either
+   */
+  def fromEither[E, A](either: Either[E, A]): ZQuery[Any, E, A] =
+    fromEffect(ZIO.fromEither(either))
```

Comment:
I think this is going to be a lot more efficient if we implement it directly instead of going through `ZIO`.

Reply:
Done, though changed to use by name parameters and lifted the call into `succeed` to mirror ZIO's implementation and avoid user surprises.